### PR TITLE
Delete CLabelsFactory

### PR DESCRIPTION
**Pull request by shubham808**
This completes #4277

### CODE REVIEW

File: doc/ipython-notebooks/structure/multilabel_structured_prediction.ipynb [hunk 1]
```diff
@@ -266,7 +266,7 @@
     "    \n",
     "    test_labels = create_labels(Y_test, n_classes)\n",
     "    \n",
-    "    out_labels = LabelsFactory.to_structured(machine.apply(feats_test))\n",
```
Comment (karlnapf) [inline doc/ipython-notebooks/structure/multilabel_structured_prediction.ipynb]: does this work without further downstream changes?

Comment (shubham808) [reply doc/ipython-notebooks/structure/multilabel_structured_prediction.ipynb]: see [here](https://github.com/shogun-toolbox/shogun/blob/986f97ee9d25898b8a3f75fa9976cc55305b4fdc/examples/undocumented/python/structure_hierarchical_multilabel_classification.py#L124) this is a similar situation and the example compiles correctly

File: examples/undocumented/python/structure_multiclass_bmrm.py [hunk 2]
```diff
@@ -65,11 +65,11 @@ def structure_multiclass_bmrm(fm_train_real=traindat,label_train_multiclass=labe
 	sosvm.set_verbose(True)
 	sosvm.train()
 
-	ppbmrm_out = LabelsFactory.to_multiclass_structured(sosvm.apply())
+	ppbmrm_out = sosvm.apply()
 	count = 0
 	for i in range(ppbmrm_out.get_num_labels()):
-		yi_pred = RealNumber.obtain_from_generic(ppbmrm_out.get_label(i))
-		if yi_pred.value == label_train_multiclass[i]:
+		yi_pred = ppbmrm_out.get_real_vector("labels_vector")[i]
```
Comment (karlnapf) [inline examples/undocumented/python/structure_multiclass_bmrm.py]: is this registered now?

File: src/shogun/structure/MulticlassSOLabels.cpp [hunk 3]
```diff
@@ -101,6 +101,9 @@ void CMulticlassSOLabels::init()
 			MS_NOT_AVAILABLE);
 	SG_ADD(&m_num_labels_set, "m_num_labels_set", "The number of assigned labels",
 			MS_NOT_AVAILABLE);
+	SG_ADD(
```
Comment (karlnapf) [inline src/shogun/structure/MulticlassSOLabels.cpp]: ah here we are ! :)

Comment (shubham808) [reply src/shogun/structure/MulticlassSOLabels.cpp]: yes :+1:

File: doc/ipython-notebooks/structure/multilabel_structured_prediction.ipynb (hunk 1 above)
Comment (karlnapf) [reply doc/ipython-notebooks/structure/multilabel_structured_prediction.ipynb]: i see so it is just passed on. Cool! :)

**Comment by karlnapf** [comment]
Great clean up patch!

**Pull request merged**